rust/datafusion/src/optimizer/utils.rs

```diff
@@ -278,6 +280,8 @@ pub fn expr_sub_expressions(expr: &Expr) -> Result<Vec<Expr>> {
         Expr::ScalarVariable(_) => Ok(vec![]),
         Expr::Not(expr) => Ok(vec![expr.as_ref().to_owned()]),
         Expr::Sort { expr, .. } => Ok(vec![expr.as_ref().to_owned()]),
+        // Between is rewritten in the physical plan
+        Expr::Between { .. } => Ok(vec![]),
```

Comment:
I think this should be implement ed as well?

Reply:
See comment above. You are probably correct though.